src/platforms/common/aux_serial.c

```diff
@@ -339,7 +339,8 @@ size_t aux_serial_transmit_buffer_fullness(void)
 void aux_serial_switch_transmit_buffers(void)
 {
 	/* Make the buffer we've been filling the active DMA buffer, and swap to the other */
-	dma_set_memory_address(USBUSART_DMA_BUS, USBUSART_DMA_TX_CHAN, (uintptr_t)aux_serial_current_transmit_buffer());
+	const char *current_buffer = aux_serial_current_transmit_buffer();
+	dma_set_memory_address(USBUSART_DMA_BUS, USBUSART_DMA_TX_CHAN, (uint32_t)current_buffer);
```

Comment:
If memory serves, this should still cast to `uintptr_t` and then naturally decay to `uint32_t` as the pointer model requires all valid pointer casts to be to either `intptr_t` or `uintptr_t`.

This looks good otherwise!